azkaban-common/src/main/java/azkaban/executor/ExecutorApiClient.java

```diff
@@ -174,16 +182,53 @@ public boolean isReverseProxyEnabled() {
    *
    * @return http client
    */
-  @Override
-  protected CloseableHttpClient createHttpClient() {
-    if (!this.isTlsEnabled) {
-      return super.createHttpClient();
-    }
+  protected CloseableHttpClient createHttpsClient() {
     final HttpClientBuilder httpClientBuilder = HttpClients.custom()
         .setSSLSocketFactory(this.tlsSocketFactory);
     return httpClientBuilder.build();
   }
 
+  /**
+   * function to perform a Post https request.
+   *
+   * @param uri    the URI of the request.
+   * @param params the form params to be posted, optional.
+   * @return the response object type of which is specified by user.
+   * @throws UnsupportedEncodingException, IOException.
+   */
+  public String httpsPost(final URI uri, final List<Pair<String, String>> params)
+          throws IOException {
+    // shortcut if the passed url is invalid.
+    if (null == uri) {
+      logger.error(" unable to perform httpPost as the passed uri is null.");
+      return null;
+    }
+
+    final HttpPost post = new HttpPost(uri);
+    return this.sendAndReturnHttps(completeRequest(post, params));
+  }
+
+  public String doPost(final URI uri, final DispatchMethod dispatchMethod, final List<Pair<String, String>> params)
+          throws IOException {
+    // If in future tls support is added for POLL based model, then following condition
+    // can be simplified
+    if (isTlsEnabled && null != dispatchMethod && dispatchMethod == DispatchMethod.CONTAINERIZED) {
```

Comment:
For any dispatch method, if Tls is enabled then use https and if not then http. Shouldn't that be the only check?

Reply:
So we have ramp-up when containerization is enabled which makes different executions have different dispatch methods. TLS is enabled only for containerized execution while the property to enable TLS will continue to be true, we still won't be using TLS for POLL-based executions until the TLS is also supported for POLL-based dispatch. Hence we needed extra check.

Reply:
Oh got it now. Thank you for clarifying that.